lib/build.rb

```diff
@@ -34,10 +34,41 @@ def initialize(token, repository, max_age)
     github_check_rate_limit(@client.last_response.headers)
   end
 
-  def query_branches
+  def query_for_new_builds
+    # Keep a mapping of branch name -> PR number for PRs which are based on non-fork branches
+    # That way we can try to comment on the PRs for any associated PR, whether internal or external
+    # The only time we will have to revert to commenting on a commit itself is if it is a non-fork branch without a PR
+    branch_to_pr_number = {}
+
+    # Thiu section scans for 'external' pull_requests (PRs from repo branches are built as separate branches)
+    # This line is where we want to add :accept => 'application/vnd.github.shadow-cat-preview+json' for draft PRs
+    pull_requests = github_query(@client) { @client.pull_requests(@repository, :state => 'open', :per_page => 50) }
+    pull_requests.each do |p|
+      # TODO: p.head.repo can be null if the fork repo is deleted.  Need to protect that here.
+      if p.head.repo.nil?
+        $logger.info("Skipping potential PR (#{p.number}): Forked repo is null (deleted?)")
+      else
+        begin
+          pb = PotentialBuild.new(@client, @token, p.head.repo.full_name, p.head.sha, p.head.ref, p.head.user.login, p.number, p.base.repo.full_name, p.base.ref)
+
+          if p.head.repo.full_name == p.base.repo.full_name
+            # we want to avoid pull
+            $logger.info("Skipping pull-request originating from head repo: #{p.number}")
+            branch_to_pr_number[branch_name] = p.number
```

Comment:
Here is where I store a hash of "branch name" -> "pr number" pairs.  Previously, if Decent CI was working on a branch from NREL/EnergyPlus itself, it wouldn't know about an associated PR number.  Now we look through PRs first and store any internal branches with their PR numbers.